pkg/deploy/controller/deployerpod/controller.go

```diff
@@ -114,42 +127,33 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 
 	if currentStatus != nextStatus {
 		deployment.Annotations[deployapi.DeploymentStatusAnnotation] = string(nextStatus)
-		if _, err := c.deploymentClient.updateDeployment(deployment.Namespace, deployment); err != nil {
+		if _, err := c.kClient.ReplicationControllers(deployment.Namespace).Update(deployment); err != nil {
 			if kerrors.IsNotFound(err) {
 				return nil
 			}
 			return fmt.Errorf("couldn't update Deployment %s to status %s: %v", deployutil.LabelForDeployment(deployment), nextStatus, err)
 		}
 		glog.V(4).Infof("Updated deployment %s status from %s to %s (scale: %d)", deployutil.LabelForDeployment(deployment), currentStatus, nextStatus, deployment.Spec.Replicas)
+
+		// If the deployment was canceled, trigger a reconcilation of its deployment config
+		// so that the latest complete deployment can immediately rollback in place of the
+		// canceled deployment.
+		if nextStatus == deployapi.DeploymentStatusFailed && deployutil.IsDeploymentCancelled(deployment) {
+			// If we are unable to get the deployment config, then the deploymentconfig controller will
+			// perform its duties once the resync interval forces the deploymentconfig to be reconciled.
+			name := deployutil.DeploymentConfigNameFor(deployment)
+			if config, err := c.client.DeploymentConfigs(deployment.Namespace).Get(name); err == nil {
+				if config.Annotations == nil {
+					config.Annotations = make(map[string]string)
+				}
+				config.Annotations[deployapi.DeploymentCancelledAnnotation] = strconv.Itoa(config.Status.LatestVersion)
+				return kclient.RetryOnConflict(kclient.DefaultRetry, func() error {
+					_, err := c.client.DeploymentConfigs(config.Namespace).Update(config)
```

Comment:
Yup, good catch. That was the reason I fetched the config in the first place and didn't use the decoded one but it seems I missed this:)